crowbar_framework/app/models/api/upgrade.rb

```diff
@@ -362,15 +362,55 @@ def nodes
 
       protected
 
+      # If there's separate network cluster, we have touch it before we start upgrade of other
+      # nodes, specificaly we need to evacuate the network routers from the first network node.
+      def prepare_network_node(network_node)
+        return if network_node.upgraded?
+
+        evacuate_network_node(network_node, network_node)
```

Comment:
@rhafer Thanks, so I've updated it to locate some controller node and install `neutron-ha-tool`. So far no checking for version of such node (maybe it's enough to ensure whole `prepare_network_node` is run only once)